### PR TITLE
storage: Add ttl tombstone statistics

### PR DESCRIPTION
**Pull request by Connor1996**
### What problem does this PR solve?

Add ttl tombstone statistics so that we can know the skipped ttl kvs when scanning.

### What is changed and how it works?

Add a thread-local `TTL_TOMBSTONE`, and record the diff in `Cursor` for every action. The statistics will be flushed into scan detail metrics finnally.

### Related changes

- Need to cherry-pick to the release branch

### Check List 

Tests 

- None

### Release note 
- None

### CODE REVIEW

File: src/storage/kv/stats.rs [hunk 1]
```diff
@@ -44,7 +46,7 @@ impl CfStatistics {
         self.get + self.next + self.prev + self.seek + self.seek_for_prev
     }
 
-    pub fn details(&self) -> [(&'static str, usize); 11] {
+    pub fn details(&self) -> [(&'static str, usize); 12] {
```
Comment (hicqu) [inline src/storage/kv/stats.rs:49]: How about making `12` as a constant?

Comment (Connor1996) [reply src/storage/kv/stats.rs:49]: good idea

**Review by hicqu — APPROVED** [review]

File: components/tikv_kv/src/stats.rs [hunk 2]
```diff
@@ -16,6 +19,63 @@ const STAT_NEXT_TOMBSTONE: &str = "next_tombstone";
...
+            kind,
+            internal_tombstone: PerfContext::get().internal_delete_skipped_count() as usize,
+            ttl_tombstone: TTL_TOMBSTONE.with(|m| *m.borrow()),
+        }
+    }
+}
+
+impl Drop for StatsCollector<'_> {
+    fn drop(&mut self) {
+        self.stats.ttl_tombstone += TTL_TOMBSTONE.with(|m| *m.borrow()) - self.ttl_tombstone;
```
Comment (BusyJay) [inline components/tikv_kv/src/stats.rs:56]: I'm afraid it's wrong. The future can be yield and fetched into other thread.

Comment (Connor1996) [reply components/tikv_kv/src/stats.rs:56]: In the span of StatsCollector lifetime, we just call rocksdb iter API. We don't call a `.await` in there, so the future would not be yield.

**Review by BusyJay — APPROVED** [review]

**Pull request merged**